RE-SS3D/SS3D
# Improve asset data

## Summary

**Documentation:** https://ss3d.gitbook.io/systems/asset-data

Removes the need of `AddressableReferences` from the `AssetData` system, aswell as renaming it to `Assets`. This PR also automatizes the whole asset setup process, as the developer now only has to select which asset group he wants to create that asset list from, and click on "Load assets from addressables group", and then follow the same process as before, clicking on "Create Enum".

### Testing:
- Check if all addressable items are in the correct addressables group in the addressables's group window.,
- Check the `Project Settings` window, and see if the `Load All Asset Databases` button is functioning correctly.
- Check if all the `Asset Databases` in the project have all the assets loaded properly when pressing the `Load Assets From Addressables Group` window.
- Check if `Assets` uses are working normally on the built game.
